pkg/controllers/cloudprovideraccount/controller.go

```diff
@@ -131,9 +137,16 @@ func (r *CloudProviderAccountReconciler) processCreateOrUpdate(namespacedName *t
 	r.Log.Info("Received request", "account", namespacedName, "operation", "create/update")
 	accountCloudType, err := util.GetAccountProviderType(account)
 	if err != nil {
-		return fmt.Errorf("%s, err: %v", accountmanager.ErrorMsgAccountAddFail, err)
+		return fmt.Errorf("%s: %v", accountmanager.ErrorMsgAddOrUpdateAccount, err)
+	}
+	if err = r.AccManager.AddAccount(namespacedName, accountCloudType, account); err != nil {
+		account.Status.Error = err.Error()
```

Comment:
this will retry reconciler right?

Reply:
No. I want to capture any errors as part of AddAccount in CPA status field, so that user can check the CPA status field.

In L145, if there is any error as part of updating status field, then we will lose the error context. So we will be return error and let the reconciler retry.